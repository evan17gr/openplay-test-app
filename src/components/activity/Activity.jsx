import propTypes from 'prop-types';
import {Typography} from 'antd';

import Tag from '../common/Tag';
import Card from '../common/Card';
import CenteredSection from '../common/CenteredSection';

const {Text} = Typography;

const Activity = ({activity}) => {
  return (
    <Card title={activity?.name}>
      <CenteredSection style={{flexDirection: 'column'}}>
        <Text strong>Start time: {activity?.start_time}</Text>
        <Text strong> End time: {activity?.end_time}</Text>
        <Text strong>Level: {activity?.level}</Text>
        <Text strong>Location: {activity?.location}</Text>
        <CenteredSection style={{margin: '0.5rem 0'}}>
          {activity?.tags && activity.tags.map((tag) => <Tag key={tag?.id} tagName={tag?.name} />)}
        </CenteredSection>
      </CenteredSection>
    </Card>
  );
};

Activity.propTypes = {
  activity: propTypes.shape({
    name: propTypes.string.isRequired,
    id: propTypes.number.isRequired,
    start_time: propTypes.string.isRequired,
    end_time: propTypes.string.isRequired,
    level: propTypes.string.isRequired,
    location: propTypes.string.isRequired,
    facility_id: propTypes.number.isRequired,
    tags: propTypes.arrayOf(
      propTypes.shape({
        id: propTypes.number.isRequired,
        name: propTypes.string.isRequired,
      }).isRequired,
    ),
  }).isRequired,
};

export default Activity;
