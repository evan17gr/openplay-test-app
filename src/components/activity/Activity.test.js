import {render} from '@testing-library/react';
import '@testing-library/jest-dom';

import Activity from '../activity/Activity';

const exampleProps = {
  id: 1,
  name: 'Diving',
  start_time: '9:00',
  end_time: '10:00',
  level: 'Experienced',
  location: 'Swimming pool 2',
  facility_id: 84,
  tags: [
    {
      id: 540,
      name: 'Swimming',
    },
  ],
};

describe('Activity', () => {
  test('is rendered', () => {
    const {container} = render(<Activity />);
    expect(container).toBeInTheDocument();
  });

  test('Check props', () => {
    const component = render(<Activity {...exampleProps} />);
    expect(component).toMatchSnapshot();
  });
});
