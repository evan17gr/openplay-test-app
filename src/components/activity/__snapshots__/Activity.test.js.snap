// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Check props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-card ant-card-bordered ant-card-hoverable"
        style="margin: 1rem 0px;"
      >
        <div
          class="ant-card-cover"
        >
          <div
            class="ant-image"
            style="width: 300px;"
          >
            <img
              class="ant-image-img"
              src="https://gw.alipayobjects.com/zos/antfincdn/aPkFc8Sj7n/method-draw-image.svg"
            />
            <div
              class="ant-image-mask"
            >
              <div
                class="ant-image-mask-info"
              >
                <span
                  aria-label="eye"
                  class="anticon anticon-eye"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="eye"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                    />
                  </svg>
                </span>
                Preview
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <section
            style="display: flex; justify-content: center; align-items: center;"
          >
            <section
              style="display: flex; justify-content: center; align-items: center; flex-direction: column;"
            >
              <span
                class="ant-typography"
              >
                <strong>
                  Start time: 
                </strong>
              </span>
              <span
                class="ant-typography"
              >
                <strong>
                   End time: 
                </strong>
              </span>
              <span
                class="ant-typography"
              >
                <strong>
                  Level: 
                </strong>
              </span>
              <span
                class="ant-typography"
              >
                <strong>
                  Location: 
                </strong>
              </span>
              <section
                style="display: flex; justify-content: center; align-items: center; margin: 0.5rem 0px;"
              />
            </section>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-card ant-card-bordered ant-card-hoverable"
      style="margin: 1rem 0px;"
    >
      <div
        class="ant-card-cover"
      >
        <div
          class="ant-image"
          style="width: 300px;"
        >
          <img
            class="ant-image-img"
            src="https://gw.alipayobjects.com/zos/antfincdn/aPkFc8Sj7n/method-draw-image.svg"
          />
          <div
            class="ant-image-mask"
          >
            <div
              class="ant-image-mask-info"
            >
              <span
                aria-label="eye"
                class="anticon anticon-eye"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="eye"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                  />
                </svg>
              </span>
              Preview
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <section
          style="display: flex; justify-content: center; align-items: center;"
        >
          <section
            style="display: flex; justify-content: center; align-items: center; flex-direction: column;"
          >
            <span
              class="ant-typography"
            >
              <strong>
                Start time: 
              </strong>
            </span>
            <span
              class="ant-typography"
            >
              <strong>
                 End time: 
              </strong>
            </span>
            <span
              class="ant-typography"
            >
              <strong>
                Level: 
              </strong>
            </span>
            <span
              class="ant-typography"
            >
              <strong>
                Location: 
              </strong>
            </span>
            <section
              style="display: flex; justify-content: center; align-items: center; margin: 0.5rem 0px;"
            />
          </section>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
